### PR TITLE
3192 remove subject and metric data classes

### PR DESCRIPTION
**Pull request by emdv88**

### CODE REVIEW

File: components/server/src/external/database/reports.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ def _prepare_documents_for_insertion(
             document[key] = value
 
 
-def latest_report(database: Database, data_model, report_uuid: str) -> Report | None:
+def latest_report(database: Database, data_model, report_uuid: str):
```
Comment (fniessink) [inline components/server/src/external/database/reports.py:105]: Why did you remove the return type?

File: components/server/tests/external/routes/test_notification.py [hunk 2]
```diff
@@ -22,6 +22,7 @@ def setUp(self):
         self.database = Mock()
         self.report = Report({}, create_report())
         self.database.reports.find.return_value = [self.report]
```
Comment (fniessink) [inline components/server/tests/external/routes/test_notification.py]: Can this line be removed?

**Review by fniessink — APPROVED** [review]

File: components/server/src/external/database/reports.py (hunk 1 above)
Comment (emdv88) [reply components/server/src/external/database/reports.py:105]: Because otherwise mypy starts complaining about bad typing in all endpoints where this function is used without covering the case that the report is None (report_uuid doesnt exist, these are about 15 different endpoints for reports and notifications).

The solution is of course to implement a 404 error because the specified report has not been found. However, I think this should fall outside of the scope of this story. Instead I made story #3717 to solve this.

File: components/server/tests/external/routes/test_notification.py (hunk 2 above)
Comment (emdv88) [reply components/server/tests/external/routes/test_notification.py]: removed, good find!

File: components/server/src/external/database/reports.py (hunk 1 above)
Comment (fniessink) [reply components/server/src/external/database/reports.py:105]: Ah, clear.

**Review by fniessink — APPROVED** [review]
Don't forget to fix the SonarQube code smell.

**Comment by emdv88** [comment]
Rebased and fixed the code smell. I'll merge as soon as the pipeline passed
